firedrake/petsc.py

```diff
@@ -77,6 +79,24 @@ def munge(keys):
     return new
 
 
+@functools.lru_cache()
+def get_petsc_variables():
+    """Get dict of PETSc environment variables from the file:
+    $PETSC_DIR/$PETSC_ARCH/lib/petsc/conf/petscvariables
+
+    The result is memoized to avoid constantly reading the file.
+    """
+    config = petsc4py.get_config()
+    path = [config["PETSC_DIR"], config["PETSC_ARCH"], "lib/petsc/conf/petscvariables"]
+    variables_path = os.path.join(*path)
+    with open(variables_path) as fh:
+        # Split lines on '=', assignment
+        splitlines = (line.split("=") for line in fh.readlines())
```

Comment:
```
var, val = line.split("=", maxsplit=1)
```

Reply:
Now fixed, but used:

`splitlines = (line.split("=", maxsplit=1) for line in fh.readlines())`

for a neater dictionary comprehension in the return line. There is no need to split into `var` and `val` as these values are never used separately.